src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -517,15 +517,26 @@ private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase)
      * @param lineNo The line number in the file.
      * @return True if a match was found inside a comment.
      */
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
-    @SuppressWarnings("deprecation")
     private boolean matchesComment(Pattern pattern, int lineNo) {
         final String line = getLine(lineNo - 1);
```

Comment:
it was our bad to put no much detail in issue, but we need to make Check to be AST based, not based on raw line content.
I update issue.
Please update Check to be comments aware and remove use of `getLine()` also.

Reply:
done